components/admin_console/cloud_import/components/open_file.tsx

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef, useState, Dispatch, SetStateAction} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import CardContainer from 'components/common/card_container';
+import {useDragAndDrop} from 'components/common/hooks/drag_and_drop';
+import FileUploadOverlay from 'components/file_upload_overlay';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+import loadingIcon from 'images/cloud_spinner.gif';
+
+import ChatServiceLogoCard from '../widgets/chat_service_logo_card';
+
+import UpgradeLink from 'components/widgets/links/upgrade_link';
+
+import {
+    CloudLinks,
+} from 'utils/constants';
+
+import OpenFileImg from '../open_file.svg';
+
+import './open_file.scss';
+import FilePreview from 'components/file_preview/file_preview';
+import AlertBanner from 'components/alert_banner';
+
+export interface StepOneProps {
+    chatService: string;
+    nextStepFunc: () => void;
+    maxFreeUsers: number;
+}
+
+const StepOne: React.FC<StepOneProps> = (props: StepOneProps) => {
+    const inputFileRef = useRef<HTMLInputElement>(null);
+    const chatService = props.chatService;
+    const [fileValues, setFileValues] = useState('initial');
+    const [showOverlay, setShowOverlay] = useState(false);
+
+    // TODO: take this info from subscription stats
+    const [isPaidTier] = useState(false);
+    const [deactivateUsers, setDeactivateUsers] = useState(false);
+
+    enum FileState {
+        INITIAL,
+        LOADING,
+        COMPLETED
+    }
+
+    const [openFileState, setOpenFileState] = useState(FileState.INITIAL);
+
+    const handleDropCb = (e: any) => {
+        // TODO: TEMPORAL - validating and parsing a simple JSON file
+        const file = e.dataTransfer.files[0];
+        onFileLoaded(file);
+        setShowOverlay(false);
+    };
+
+    const handleDragOverCb = () => {
```

Comment:
For readability, I would suggest `Cb` here on the naming. Doesn't provide any value and the `handle` naming convention may be explicit enough.
```suggestion
    const handleDragOver = () => {
```